mne/preprocessing/ica.py

```diff
@@ -403,12 +406,16 @@ def _fit_epochs(self, epochs, picks, decim, verbose):
             self.max_pca_components = len(picks)
             logger.info('Inferring max_pca_components from picks.')
 
+        # this should be a copy (picks a list of int)
         data = epochs.get_data()[:, picks]
+        # this will be a view
         if decim is not None:
-            data = data[:, :, ::decim].copy()
+            data = data[:, :, ::decim]
```

Comment:
```Python
import time
from mne.io import Raw
from mne.preprocessing import ICA
from mne.datasets import sample
data_path = sample.data_path()
raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
raw = Raw(raw_fname, preload=True)
raw.pick_types(meg=True, exclude='bads')
t0 = time.time()
ica = ICA(method='fastica', random_state=0).fit(raw, decim=4)
print(time.time() - t0)
```
master: ~15.3 sec
PR: ~14.6 sec

Based on my CPU usage it still ends up being SIMD because of the copy that (sometimes) happens in `self._pre_whiten`. Is the heavy lifting done in `self._fit`? Maybe that should do `np.ascontiguousarray` to ensure we still get SIMD capability.